onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1288,26 +1288,24 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<Node*>& fuse
           shape_ranges = DeserializeProfile(profile_file);
           LOGS_DEFAULT(VERBOSE) << "[TensorRT EP] DeSerialized " + profile_cache_path;
           // Deserialize engine
-          trt_state->context->reset();
-          trt_state->engine->reset();
+          trt_state->context = nullptr;
```

Comment:
this is problematic. previous reset will free the memory, so directly setting to nullptr will cause mem leak. 
but if you free it here, that is also a problem because technically this object is owned by the unique_ptr.
need to rethink this if we want to refactor.